### DISCUSSION TITLE
exchange-multipart-fetch compatibility with urql@2

### DISCUSSION THREAD

**Discussion by artecoop** [urql-graphql/urql#1407] [Q&A] [answered]
Hello everyone!
@urql/exchange-multipart-fetch seems to be incompatible with urql v2. There is a timeline for this component to be aligned with current build of urql?

Thanks!

**Comment by JoviDeCroock** [comment, marked as answer]
What is the problem you're facing? The tests still pass so is this a React-specific issue? Try to explain a bit of detail concerning your problem, it's hard for us to guess what you're running into.

**Reply by artecoop** [reply]
I got this error (I'm using typescript)

```
Type 'import("/home/admin/node_modules/@urql/core/dist/types/types").Exchange' is not assignable to type 'import("/home/admin/node_modules/urql/node_modules/@urql/core/dist/types/types").Exchange'.
  Types of parameters 'input' and 'input' are incompatible.
```
I've already clean up node_modules to avoid problems

**Reply by artecoop** [reply]
Well, out of curiosity, I deleted both package-lock.json and node_modules/ and installed again with yarn 1.22.10 instead of using npm 7.5.4. Well the error is gone. Sorry to bother you with something not related to this package itself

**Reply by JoviDeCroock** [reply]
Oh this often occurs with dependencies that relate under the hood, when you run into this issue you'll have to run [`yarn-deduplicate`](https://github.com/atlassian/yarn-deduplicate) or [`npm-dedupe`](https://docs.npmjs.com/cli/v6/commands/npm-dedupe/). This happens when your lockfile thinks that the dependency relies on a different version of the sub-dependency making you have `@urql/core-v1` used in the exchange and `@urql/core-v2` in your actual code.

**Reply by artecoop** [reply]
Thanks for this suggestion. I'll pipeline this command on my update routine.
